projects/fsastorefrontlib/src/cms-components/product/product-list/product-list.component.ts

```diff
@@ -1,8 +1,44 @@
 import { Component } from '@angular/core';
-import { ProductListComponent } from '@spartacus/storefront';
+import { RoutingService } from '@spartacus/core';
+import {
+  PageLayoutService,
+  ProductListComponent,
+  ProductListComponentService,
+  ViewConfig,
+} from '@spartacus/storefront';
+import { FSProduct } from './../../../occ/occ-models/occ.models';
+import { FSCheckoutConfigService } from '../../../core/checkout/services/checkout-config.service';
 
 @Component({
   selector: 'cx-fs-product-list',
   templateUrl: './product-list.component.html',
 })
-export class FSProductListComponent extends ProductListComponent {}
+export class FSProductListComponent extends ProductListComponent {
+  constructor(
+    pageLayoutService: PageLayoutService,
+    productListComponentService: ProductListComponentService,
+    scrollConfig: ViewConfig,
+    protected checkoutConfigService: FSCheckoutConfigService,
+    protected routingService: RoutingService
+  ) {
+    super(pageLayoutService, productListComponentService, scrollConfig);
+  }
+
+  protected categoryBasedSteps = ['chooseCoverStep', 'comparisonCheckoutStep'];
+
+  startCheckoutForProduct(product: FSProduct) {
+    let routingParam;
+    const initialStep = this.checkoutConfigService.getInitialStepForCategory(
+      product.defaultCategory.code
+    );
+    if (this.categoryBasedSteps.includes(initialStep.id)) {
+      routingParam = product.defaultCategory.code;
+    } else {
+      routingParam = product.code;
+    }
```

Comment:
Maybe you could use here ternary operator:

```
const routingParam = { 
     code: this.categoryBasedSteps.includes(initialStep.id) ? product.defaultCategory.code : product.code
 }
```

And when you call routing service instead of `params: { code: routingParam }` it's enough to pass just `routingParam`

Reply:
No need, the same effect is achieved with this one. Also, if there is a new use case (initial step that is not dependant on product code or category) then we would need to add new "else if" and ternary operator would be removed.